lib/srv/desktop/rdp/rdpclient/src/client.rs

```diff
@@ -121,22 +143,27 @@ impl Client {
             read_stream: Some(read_stream),
             write_stream: Some(write_stream),
             x224_processor: Some(Arc::new(Mutex::new(x224_processor))),
-            write_requester: None,
-            function_receiver: None,
+            client_handle: Some(client_handle),
+            function_receiver: Some(function_receiver),
         })
     }
 
     /// Registers the Client with the [`global::CLIENT_HANDLES`] cache.
     ///
-    /// This constitutes creating a new [`ClientHandle`]/[`FunctionReceiver`] pair,
-    /// storing the [`ClientHandle`] (indexed by `self.cgo_handle`) in [`global::CLIENT_HANDLES`],
-    /// and assigning the [`FunctionReceiver`] to `self.function_receiver`.
-    fn register(mut self) -> Self {
-        let (client_handle, function_receiver) = channel(100);
-        self.write_requester = Some(client_handle.clone());
-        self.function_receiver = Some(function_receiver);
-        global::CLIENT_HANDLES.insert(self.cgo_handle, client_handle);
-        self
+    /// This constitutes storing the [`ClientHandle`] (indexed by `self.cgo_handle`)
+    /// in [`global::CLIENT_HANDLES`].
+    fn register(self) -> ClientResult<Self> {
+        // These should have been initialized in [`Self::connect`].
+        if self.client_handle.is_none() || self.function_receiver.is_none() {
+            return Err(ClientError::InternalError);
+        }
+
+        global::CLIENT_HANDLES.insert(
+            self.cgo_handle,
+            // unwrap is safe because we just checked that it is Some
+            self.client_handle.as_ref().unwrap().clone(),
+        );
```

Comment:
Consider 
```suggestion
        // These should have been initialized in [`Self::connect`].
        let client_handle = self
            .client_handle
            .as_ref()
            .ok_or(ClientError::InternalError)?;
        self.function_receiver
            .as_ref()
            .ok_or(ClientError::InternalError)?;

        global::CLIENT_HANDLES.insert(self.cgo_handle, client_handle.clone());
```
`unwrap` makes me nervous

Reply:
Fair enough, how about just this: c0f72f28d405b6b457d29bcaa2d482518228d4f1